src/pages/index.tsx

```diff
@@ -8,21 +10,54 @@ import task from '@/interfaces/task.type';
 import Accordion from '@/components/Accordion';
 
 const TASKS_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/tasks`;
+const SELF_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/users/self`;
 
-function renderCardList(tasks: task[]) {
-  return tasks.map((item: task) => <Card content={item} key={item.id} />);
+async function updateCardContent(cardDetails: any) {
+  let response = {};
+  let error = {};
+  try {
+    response = await axios({
```

Comment:
Use our own [fetch helper function](https://github.com/Real-Dev-Squad/website-status/blob/develop/src/helperFunctions/fetch.ts)

Reply:
Do you mean like this?
```async function updateCardContent(cardDetails: task) {
    const responsePromise = await axios({
      method: 'patch',
      url: `https://api.realdevsquad.com/tasks/${cardDetails.id}`,
      params: null,
      data: cardDetails,
      headers: {
        'Content-type': 'application/json',
      },
      withCredentials: true,
    });
    return responsePromise;

Reply:
Do you mean like this?
```async function updateCardContent(cardDetails: task) {
  try {
    const response = await fetch({
      url: `${TASKS_URL}/${cardDetails.id}`,
      method: 'patch',
      params: null,
      data: cardDetails,
      headers: {
        'Content-type': 'application/json',
      },
    });
    // eslint-disable-next-line no-console
    console.log('Response', response);
  } catch (err:any) {
    // eslint-disable-next-line no-console
    console.log(toast);
  }
}

Reply:
Check how we are using this helper function in other places in the code base

Reply:
Check how we are using this helper function in other places in the code base